content/academy/puppeteer_playwright/browser.md

```diff
@@ -0,0 +1,90 @@
+import { chromium } from 'playwright';
+
+await chromium.launch();
+</marked-tab>
+```
+
+When we run this code with the command `node index.js`, a browser will open up; however, we won't actually see anything. This is because the default mode of a browser after `launch()`ing it is **headless**, meaning that it has no visible UI.
+
+> If you run this code right now, it will hang. Use **control^** + **C** to force quit the program.
+
+## [](#launch-options) Launch options
+
+In order to see what's actually happening, we can pass an **options** object ([Puppeteer](https://pptr.dev/#?product=Puppeteer&version=v13.7.0&show=api-puppeteerlaunchoptions), [Playwright](https://playwright.dev/docs/api/class-browsertype#browser-type-launch)) with **headless** set to **false**.
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+await chromium.launch({ headless: false });
+</marked-tab>
+<marked-tab header="Puppeteeer" lang="javascript">
+import { chromium } from 'playwright';
+
+await chromium.launch({ headless: false });
+</marked-tab>
+```
+
+> Playwright also supports Firefox and Webkit (Safari), while Puppeteer only supports Chromium based browsers. For ease of understanding, we've chosen to use Chromium in the Playwright examples to keep things working on the same plane.
+
+Now we'll actually see a browser open up.
+
+![Chromium browser opened by Puppeteer/Playwright]({{@asset puppeteer_playwright/images/chromium.webp}})
+
+There are a whole lot more options that we can pass into the `launch()` function, which we'll be getting into a little bit later on.
+
+## [](#browser-methods) Browser methods
+
+Not only does the `launch()` function open up a browser for us, but it also returns an object representation of the browser that can be manipulated and used to interact with it in our code. This **Browser** object ([Puppeteer](https://pptr.dev/#?product=Puppeteer&version=v13.7.0&show=api-class-browser), [Playwright](https://playwright.dev/docs/api/class-browser)) has many functions which make it easy to do so. One of these functions is `close()`. Until now, we've been using **control^** + **C** to force quit the process, but with this function, we'll no longer have to do that.
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+const browser = await chromium.launch({ headless: false });
+
+// code will be here in the future
+
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteeer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+const browser = await puppeteer.launch({ headless: false });
+
+// code will be here in the future
+
+await browser.close();
+</marked-tab>
+```
+
+In a few lessons from now, we'll be discussing the `browser.contexts()` (Playwright)/`browser.browserContexts()` (Puppeteer) functions and how to create **browser contexts** with another function on the **Browser** object.
```

Comment:
This feels a bit out of place. At this moment, the readers don't know what the contexts are, so the functions are not very meaningful to them.